xcell/cls/theory.py

```diff
@@ -157,6 +196,9 @@ def get_ccl_tkka(self, ccl_trA1, ccl_trA2, ccl_trB1, ccl_trB2,
         return tkk
 
     def get_ccl_pk(self, ccl_tr1, ccl_tr2):
+        self.get_cosmo_ccl()
+        self.get_halomodel_params()
```

Comment:
Same as with `get_cosmo_ccl`. It might make sense to use `hm_par = self.get_halomodel_params()` and store it internally in `self._hm_par`

Reply:
Done